File: src/solutions/python/TwoSum.py

'''
 # Problem
 	- `Link`: https://leetcode.com/problems/two-sum/
 # Solution
 	- `Author`: Kyungtaek Lim (Jonas)
 	- `Date`: Feb 15, 2025
 	- `Answer`: twoSum
'''

from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dict_temp = {}
        for index, num in enumerate(nums):
            if num in dict_temp:
                return [dict_temp.get(num), index]
            dict_temp[target - num] = index
        

solution = Solution()
if __name__ == "__main__":
    print(solution.twoSum(nums=[2, 7, 11, 15], target=9))

File: src/solutions/typescript/TwoSum.ts
/*
 # Problem
 	- `Link`: https://leetcode.com/problems/two-sum/
 # Solution
 	- `Author`: Kyungtaek Lim (Jonas)
 	- `Date`: May 26, 2025
 	- `Answer`: twoSum
*/
function twoSum(nums: number[], target: number): number[] {
    const map = new Map<number, number>();
    let n = nums.length;
    for (let i = 0; i < n; i++) {
        let num = nums[i];
        if (map.has(num)) {
            return [map.get(num)!, i];
        }
        map.set(target - num, i);
    }
    return [];
};

console.log(twoSum([2, 7, 11, 15], 9)); // [0, 1]
console.log(twoSum([3, 2, 4], 6));      // [1, 2]
console.log(twoSum([3, 3], 6));         // [0, 1]
console.log(twoSum([1, 2, 3], 10));     // []